### PR TITLE
Add a Jenkinsfile

### PR DESCRIPTION
**Pull request by dm0-**
This also makes `configure` test `gmock` and `gtest` development files.  They don't look like they are intended to be used outside tests, but a header is always included which fails the regular build if they're not present, so always require them.

### CODE REVIEW

**Comment by marineam** [comment]
What non-unit test thing is including gtest or gmock? That seems like a bug.

Adding to configure seems fine regardless.

File: configure.ac [hunk 1]
```diff
@@ -85,6 +85,22 @@ PKG_CHECK_MODULES([DEPS],
                    libxml-2.0
                    protobuf])
 
+AC_PATH_TOOL([GTEST_CONFIG], [gtest-config])
+AC_ARG_VAR([GTEST_CPPFLAGS], [CPPFLAGS for compiling with gtest])
+AC_ARG_VAR([GTEST_LIBS], [LIBS for linking with gtest])
+test -n "$GTEST_CPPFLAGS" || GTEST_CPPFLAGS=$($GTEST_CONFIG --cppflags)
```
Comment (marineam) [inline configure.ac]: I forget, does AC_PATH_TOOL abort with an error if the tool is not found? If not we should explicitly die before attempting to run the command. Alternatively if we can fix the default build (without unit tests) to work without gtest/gmock being installed then configure can gracefully skip this when the config tool isn't found.

**Comment by dm0-** [comment]
The file `src/update_engine/certificate_checker.cc` for `libupdate_engine` includes `src/update_engine/certificate_checker.h` which includes `<gtest/gtest_prod.h>` (apparently only for `FRIEND_TEST`).

**Comment by marineam** [comment]
Ohhhh, right. Forgot about that. So as long as AC_PATH_TOOL exits configure with a sensible error when gtest isn't found then this change is good.

**Comment by dm0-** [comment]
I've copied the `AS_IF` method of aborting used earlier in the file.

**Pull request merged**